pkg/middleware/recovery/recovery_test.go

```diff
@@ -0,0 +1,210 @@
+package recovery
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/test/bufconn"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/logger"
+)
+
+func TestPanic(t *testing.T) {
+	logger := logger.MustNewLogger("text", "info", "unix")
+	handlerFunc := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		panic("Unexpected error!")
+	})
+
+	t.Run("With panic Recovery", func(t *testing.T) {
+		handler := Panic(handlerFunc, logger)
+
+		req, err := http.NewRequest(http.MethodGet, "/", nil)
+		if err != nil {
+			require.NoError(t, err)
+		}
+
+		resp := httptest.NewRecorder()
+		handler.ServeHTTP(resp, req)
+
+		require.Equal(t, http.StatusInternalServerError, resp.Code)
+		require.Equal(t, http.StatusText(http.StatusInternalServerError), strings.Trim(resp.Body.String(), "\n"))
+	})
+}
+
+func TestUnaryPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
+
+	serverOpts := []grpc.ServerOption{
+		grpc.ChainUnaryInterceptor(
+			[]grpc.UnaryServerInterceptor{
+				UnaryPanicInterceptor(logger.MustNewLogger("text", "info", "unix")),
+			}...,
+		),
+	}
+
+	t.Run("With Unary Panic Interceptor", func(t *testing.T) {
```

Comment:
In this test and in the streaming test below, where are we panic'ing to test that our panic middleware is working?

I don't see the panic.

Reply:
```

func (unimplementedOpenFGAServiceServer) Check(context.Context, *openfgav1.CheckRequest) (*openfgav1.CheckResponse, error) {
	panic("Unexpected error!")
}
func (unimplementedOpenFGAServiceServer) StreamedListObjects(m *openfgav1.StreamedListObjectsRequest, stream openfgav1.OpenFGAService_StreamedListObjectsServer) error {
	_ = stream.RecvMsg(m)

	panic("Unexpected error!")
}
```